executor/write.go

```diff
@@ -114,11 +114,11 @@ func updateRecord(ctx context.Context, h int64, oldData, newData []types.Datum,
 	}
 
 	if handleChanged {
-		err = t.RemoveRecord(ctx, h, oldData)
+		_, err = t.AddRecord(ctx, newData)
 		if err != nil {
 			return false, errors.Trace(err)
 		}
-		_, err = t.AddRecord(ctx, newData)
+		err = t.RemoveRecord(ctx, h, oldData)
```

Comment:
Why change this order？

Reply:
Gotcha.

Reply:
When update conflicts, updateRecord function return an error and the RemoveRecord have taken effects. In the "UpdateIgnore" statement,  it should AppendWarning instead of generating an error. But in the "auto-increment" connection, the transaction would be commit with "RemoveRecord" but "AddRecord" Conflicts.